sqlglot/parser.py

```diff
@@ -1346,6 +1346,32 @@ def _parse_command(self) -> exp.Command:
             exp.Command, this=self._prev.text.upper(), expression=self._parse_string()
         )
 
+    def _try_parse(
+        self, parse_method: t.Callable, retreat: bool = False
+    ) -> t.Optional[exp.Expression]:
```

Comment:
Is it possible to improve the type hint here so that the return type is the same as `parse_method`'s?